Sources/MapboxNavigation/NavigationViewportDataSource.swift

```diff
@@ -0,0 +1,424 @@
+import MapboxMaps
+import MapboxCoreNavigation
+import Turf
+import MapboxDirections
+
+/**
+ Class, which conforms to `ViewportDataSource` protocol and provides default implementation of it.
+ */
+public class NavigationViewportDataSource: ViewportDataSource {
+    
+    /**
+     Delegate, which is used to notify `NavigationCamera` regarding upcoming `CameraOptions`
+     related changes.
+     */
+    public var delegate: ViewportDataSourceDelegate?
```

Comment:
Should we use `weak` here? I see possible retain loop

Reply:
Good catch, will be addressed in another PR.

Reply:
By the way, SwiftLint produces a warning for such issues.